### ISSUE TITLE
Side modal does not auto-focus

### ISSUE THREAD

**Issue by hjbrown11** [twilio-labs/paste#3275]
### Description

When I open side modal, I expect focus to move to this element, so that accessibility considerations are consistent. The Paste doc states the following:

Accessibility
Side Modal and non-modal dialogs follow these accessibility guidelines:

- There must be a focusable element inside the dialog.
- There should be a close button so screen readers have a specific close action to target.
- **A Side Modal is a focus trap, and focus is placed inside it when it's shown.**
- A Side Modal must be triggered by an explicit user action, e.g. clicking a button.

Focus does not move there automatically, however the docs seem to suggest that this would happen out of the box. 

### Link to Reproduction

https://codesandbox.io/s/pensive-sound-fd7h2p?file=/src/index.tsx

### Steps to reproduce

1. Go to the link
2. Click on the side modal button
4. See that focus isn't moved to the modal

### Paste Core Version

Latest

### Browser

Chrome latest

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by hjbrown11** [comment]
As a follow up from this, I need to implement a way to show a 'are you sure' popover when a user presses the escape key. At the moment, if a user presses the close button with the form dirty, we open an alert dialogue which asks if they want to save their changes or cancel them, however I have been unable to replicate this behaviour with the escape key so the form closes in a dirty and neither saved nor cancelled state

**Comment by SiTaggart** [comment]
Hi @hjbrown11 

Not placing user focus inside the side modal on open definitely seems like a bug. I'll try to get a fix in this week, otherwise I'll ticket it and try to slot it into an upcoming sprint.

As for the follow-up, I would suggest opening a separate discussion topic for that one so we can keep the threads clean.

Thanks

**Comment by TheSisb** [comment]
Small update: a fix will be released in the next version of Paste Core (when this PR merges: https://github.com/twilio-labs/paste/pull/3297)
